src/utilities/theme/utils.ts

```diff
@@ -1,115 +1,779 @@
-import tokens from '@shopify/polaris-tokens';
-import {HSLColor} from '../color-types';
+import {HSLColor, HSLAColor} from '../color-types';
 import {colorToHsla, hslToString, hslToRgb} from '../color-transformers';
 import {isLight} from '../color-validation';
-import {constructColorName} from '../color-names';
-import {createLightColor} from '../color-manipulation';
-import {compose} from '../compose';
-
-import {Theme, ColorsToParse, ThemeVariant, ThemeColors} from './types';
-import {needsVariantList} from './config';
-
-export function setColors(theme: Theme | undefined): string[][] | undefined {
-  let colorPairs;
-  if (theme && theme.colors) {
-    Object.entries(theme.colors).forEach(([colorKey, pairs]) => {
-      const colorKeys = Object.keys(pairs);
-      if (colorKey === 'topBar' && colorKeys.length > 1) {
-        colorPairs = colorKeys.map((key: string) => {
-          const colors = (theme.colors as ThemeColors).topBar;
-          return [constructColorName(colorKey, key), colors[key]];
-        });
-      } else {
-        colorPairs = parseColors([colorKey, pairs]);
-      }
-    });
-  }
+import {CSSProperties, Theme} from './types';
 
-  return colorPairs;
+// TODO: Pull from polaris-tokens
+enum DefaultColor {
+  Surface = '#FAFAFA',
+  OnSurface = '#1F2225',
+  InteractiveNeutral = '#EAEAEB',
+  Interactive = '#0870D9',
+  Branded = '#008060',
+  Critical = '#E32727',
+  Warning = '#FFC453',
+  Highlight = '#59D0C2',
+  Success = '#008060',
 }
 
-export function needsVariant(name: string) {
-  return needsVariantList.indexOf(name) !== -1;
-}
+export function Colors(theme: Theme) {
+  const {colors = {}} = theme;
+
+  const isLightTheme = isLight(
+    hslToRgb(colorToHsla(
+      colors == null || colors.surface == null
+        ? DefaultColor.Surface
+        : colors.surface,
+    ) as HSLColor),
+  );
```

Comment:
This is a critical piece, as it looks at the surface value and determines whether it is light or dark. Based off this, the generated color variants can respond to whether the applied theme represents light mode or dark mode.